src/repositories/account.repo.js

```diff
@@ -0,0 +1,107 @@
+import { pool } from "../config/db.js";
+import { Account } from "../models/account.model.js";
+
+export class AccountRepo {
+  static async createAccount(account) {
+    const result = await pool.query(
+      `INSERT INTO accounts (user_name, user_email, user_password) VALUES ($1, $2, $3) RETURNING *`,
+      [account.user_name, account.user_email, account.user_password]
+    );
+    if (result.rows[0].length === 0) {
+      throw new Error("Unable to create new account");
+    }
+    console.log("repo: ", account);
+    return result.rows[0];
+  }
+
+  static async findById(id) {
+    const result = await pool.query(
+      `SELECT * FROM accounts WHERE user_id = $1`,
+      [id]
+    );
+    if (result.rows.length === 0) {
+      throw new Error("Cannot find account by id: " + id);
+    }
+    return new Account(
+      result.rows[0].user_id,
+      result.rows[0].user_name,
+      result.rows[0].user_email,
+      result.rows[0].user_password
+    ).toDTO();
+  }
+
+  static async findByEmail(email) {
+    const result = await pool.query(
+      `SELECT * FROM accounts WHERE user_email = $1`,
+      [email]
+    );
+    if (result.rows.length === 0) {
+      throw new Error("Incorrect email or password");
+    }
+    return result.rows[0];
+  }
+
+  static async findAllAccounts() {
+    const result = await pool.query(`SELECT * FROM accounts`);
+    if (result.rows.length === 0) {
+      return [];
+    }
+    const allAccountsDTO = result.rows.map((account) =>
+      new Account(
+        account.user_id,
+        account.user_name,
+        account.user_email,
+        account.user_password
+      ).toDTO()
+    );
+    return allAccountsDTO;
```

Comment:
don't use naming like allAccountsDTO instead of use accounts but in this case is useless to create const

```suggestion
    return result.rows.map((account) =>
      new Account(
        account.user_id,
        account.user_name,
        account.user_email,
        account.user_password
      ).toDTO()
    );
```